router/internal/eventorder/eventorder.go

```diff
@@ -0,0 +1,255 @@
+package eventorder
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+)
+
+type State string
+
+const (
+	SucceededState State = "succeeded"
+	FailedState    State = "failed"
+	AbortedState   State = "aborted"
+	WaitingState   State = "waiting"
+)
+
+var ErrUnsupportedState error = errors.New("unsupported state")
```

Comment:
```suggestion
var ErrUnsupportedState = errors.New("unsupported state")
```

We can get rid of type declaration here